lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3SseRequestConfigurator.java

```diff
@@ -39,6 +49,47 @@ public static void addEncryptionSettings(CreateMultipartUploadRequest.Builder bu
             case NONE -> { /* ignored */ }
             case S3 -> builder.serverSideEncryption(AES256);
             case KMS -> builder.serverSideEncryption(AWS_KMS).ssekmsKeyId(context.sseKmsKeyId());
+            case CUSTOMER -> {
+                builder.sseCustomerAlgorithm(context.sseCustomerKey().algorithm());
+                builder.sseCustomerKey(context.sseCustomerKey().key());
+                builder.sseCustomerKeyMD5(context.sseCustomerKey().md5());
```

Comment:
we can use `addEncryptionSettings` here right?

Reply:
i don't get, which one?

Reply:
ignore this. I did not notice that all `addEncryptionSettings` has different builder.